client/store/cart.js

```diff
@@ -1,70 +1,111 @@
+//   cart
+// })
+
+// export const updateCart = () => {
+//   return {
+//     type: UPDATE_CART
+//   }
+// }
 
 //thunks
-export const getCartThunk = () => {
+export const getCartThunk = userId => {
   return async dispatch => {
-    const response = await axios.get('/api/cart')
-    dispatch(setCart(response.data))
+    try {
+      const {data} = await axios.get(`/api/users/cart/${userId}`)
+      dispatch(getCart(data))
+    } catch (error) {
+      console.log('Failed to get user cart', error)
+    }
   }
 }
-export const setCartThunk = cartProduct => {
-  return async dispatch => {
-    const response = await axios.post('/api/cart', cartProduct)
-    dispatch(setCart(response.data))
-  }
-}
-export const addCartThunk = (productId, userId) => {
+
+export const addToCartThunk = (productId, userId) => {
   return async dispatch => {
-    const response = await axios.post('/api/cart/add', productId, userId)
-    dispatch(setCart(response.data))
+    try {
+      const {data} = await axios.put(`/api/users/cart/${userId}`, {
+        id: productId
+      })
+      dispatch(addProduct(data))
+    } catch (error) {
+      console.log('Failed to add to cart', error)
+    }
   }
 }
-export const deleteCartProductThunk = productId => {
+
+export const deleteCartProductThunk = (product, userId) => {
   return async dispatch => {
-    const response = await axios.delete(`/api/cart/${productId}`)
-    dispatch(setCart(response.data))
+    try {
+      await axios.delete(`/api/users/cart/${userId}/${product.id}`)
+      dispatch(removeProduct(product.id))
+    } catch (error) {
+      console.log('Failed to remove item from cart', error)
+    }
   }
 }
 
 const initialState = {
-  cart: [],
-  cartTotal: 0,
-  cartSize: 0
+  products: []
 }
 
 //reducer
-export default function(state = initialState, action) {
+
+const cartReducer = (state = initialState, action) => {
   switch (action.type) {
-    case SET_CART:
-      let cartSize = 0
-      const cartTotal = action.cart.reduce((total, elem) => {
-        cartSize += elem.quantity
-        total += elem.quantity * elem.product.price
-        return total
-      }, 0)
-      return {
-        ...state,
-        cart: action.cart,
-        cartTotal,
-        cartSize
-      }
+    case GET_CART:
+      return {...state, products: [...action.products]}
+    case ADD_PRODUCT:
+      return {...state, products: [...state.products, action.product]}
+    case REMOVE_PRODUCT: {
+      let updatedCart = state.products.filter(
+        product => product.id !== action.productId
+      )
+      return {...state, products: updatedCart}
+    }
     default:
       return state
   }
 }
+
+export default cartReducer
+
+// let cartSize = 0
+// const cartTotal = action.cart.reduce((total, elem) => {
+//   cartSize += elem.quantity
+//   total += elem.quantity * elem.product.price
+//   return total
+// }, 0)
+// return {
+//   ...state,
+//   cart: action.cart,
+//   cartTotal,
+//   cartSize
+// }
```

Comment:
I'm not sure where the cart total calculation should go in the reducer, I commented this out so we don't lose it and can put it in the right place